psl-core/src/main/java/org/linqs/psl/config/Options.java

```diff
@@ -669,13 +669,44 @@
         "The alpha parameter for the dirichlet distribution of the weight sampler."
     );
 
+    public static final Option SGD_ADAM = new Option(
+            "sgd.adam",
+            false,
+            "Update learning rate according adam."
+    );
+
+    public static final Option SGD_ADA_GRAD = new Option(
+            "sgd.adagrad",
+            false,
+            "Update learning rate according ada grad."
+    );
+
+    public static final Option SGD_COORDINATE_STEP = new Option(
+            "sgd.coordinatestep",
+            false,
+            "Take coordinate decrease steps during sgd."
+    );
+
+    public static final Option SGD_INVERSE_TIME_EXP = new Option(
+        "sgd.inversescaleexp",
+        1.0f,
+        "If SGD is using the INVERSETIME schedule, then this value is used as the negative exponent of the number of iterations to scale the gradient step.",
+        Option.FLAG_POSITIVE
+    );
+
     public static final Option SGD_LEARNING_RATE = new Option(
         "sgd.learningrate",
         1.0f,
         null,
         Option.FLAG_POSITIVE
     );
 
+    public static final Option SGD_LEARNING_SCHEDULE = new Option(
+        "sgd.learningschedule",
+        "STEPDECAY",
+        "The learning schedule of the SGD inference reasoner."
```

Comment:
Is learning schedule a common term? Or is it something specific to this implementation. If it is specific, elaborate a bit more (otherwise you can ignore).